### DISCUSSION TITLE
Build + deploy pipeline for RAP application

### DISCUSSION THREAD

**Discussion by Michiel-s** [AmpersandTarski/Ampersand#1396] [General]
Inspiration from other project:

![image](https://user-images.githubusercontent.com/7961582/223091453-4c46ac69-3911-43f5-8a7b-83de9a2e46b4.png)

**Comment by FrankTeusink** [comment]
That's how I also see it. Though on RAP repository we currently have

- main branch, last commit Aug 15 2021,
- development branch, last commit Jan 17 2023,
- and a lot of other branches, of various ages.

So I think we should, for now have
- development branch as our core/master/main branch to collect all tested and approved code
- keep the real main branch as-is, and maybe discuss the intentional differences between main and development
- create feature branches based on development.

**Reply by Michiel-s** [reply]
I know how it is at this moment, but I wanted to propose a target/soll situation.

Personally I don't see use for distinction between main and development branch. In both you tested/approved code.

For smaller teams a main-branch only strategy (a.k.a. (GitHub Flow branching strategy)[https://www.flagship.io/git-branching-strategies/]) works best.

> [GitHub Flow ](https://docs.github.com/en/get-started/quickstart/github-flow)is a simpler alternative to GitFlow ideal for smaller teams as they don’t need to manage multiple versions.

Read this article about different branching strategies: https://www.flagship.io/git-branching-strategies/

In the picture I shared above, you can see that we use 'release tags' to trigger deployment to production. Main branch is always automatically deployed to a staging environment.

**Reply by FrankTeusink** [reply]
I agree. But first wanted to find out why we had separate main/development.

**Comment by FrankTeusink** [comment]
Also, for feature branches, I propose to put a prefix in the branch names of feature branches

Like
- feature/setup_GitHub_actions

That way we can auto-deploy from these branches to development environment(s)

**Comment by FrankTeusink** [comment]
So if we do GitHub flow, we need to decide what out 'main' branch is. Candidates are master and development. I see development is some 60 commits ahead of master. But also master is 2 pull requests (Aug 2021, Stef) ahead of development. So that needs some analysis and brain searching.
